.circleci/config.yml

```diff
@@ -848,8 +848,7 @@ jobs:
   # This job roughly follows the instructions from https://circleci.com/blog/publishing-to-github-releases-via-circleci/
   build_and_upload_devcontracts:
     docker:
-      # Image from https://github.com/cibuilds/github, based on alpine
-      - image: cibuilds/github:0.13
+      - image: rust:1.50.0
```

Comment:
Good change. But then we need to install ghr manually. See https://github.com/CosmWasm/wasmvm/blob/v0.14.0-alpha2/.circleci/config.yml#L146-L150

Reply:
To be continued... 🙂

Reply:
Added steps to install `ghr` (and go), but I'm now thinking that I will need to install `apk` too?

Maybe it's better to stick to the original image, and install Rust instead. Best option would be to add a separate step / image for this, but then we'll need to store the compiled contracts somewhere.

Options:

- Use rust stable image, and install required deps (ghr, go, apk, ...)
- Use original image (cibuilds/github:0.13) and install Rust stable, via rustup or something.
- Store artifacts somewhere (?), and use a different image (requires storage, multiple steps, etc.)
- Put `check_contract` into `rust-optimizer`, maybe with a `--check-contracts` or similar option.
- Remove `check_contract` from build/release step.
- ...

Reply:
I wonder why we need `check_contract` at all (here). I mean, it is nice to have, sure. But we never had problems with those Wasm files.

> but I'm now thinking that I will need to install `apk`, too?

`rust:x.y.z` is based on Debian (https://hub.docker.com/_/rust), so installing the Alpine package manager should not be required. We can install packages via `apt` instead. This should also allow us to easily install go.

> Maybe it's better to stick to the original image, and install Rust instead.

If we put the effort into this now, I like to take the opportunity to migrate away from the no-so-actively maintaned cibuilds/github.

> Best option would be to add a separate step / image for this, but then we'll need to store the compiled contracts somewhere.

The CosmJS setup shows how [to store artifacts](https://github.com/cosmos/cosmjs/blob/main/.circleci/config.yml#L64-L67) and [pick them up in another job](https://github.com/cosmos/cosmjs/blob/v0.24.0-alpha.26/.circleci/config.yml#L113-L117). However, if possible I would like to avoid splitting the job which makes the setup even more complex.

> Put check_contract into rust-optimizer, maybe with a --check-contracts or similar option.

This introduces a tight coupling between CosmWasm versions and ruther-optimizer. So far tooling in rust-optimizer is super generic and hardly changes. One version of rust-optimizer can be used across a big range of cosmwasm versions. I think this is a good thing.

Reply:
I have now solved it (I think) by running `check_contract` into its own (sub) container. Talk about bootstrapping... :-)

I'm totally OK with removing this step, though. Now that it is working we can leave it, and see how it goes, though.

Reply:
> If we put the effort into this now, I like to take the opportunity to migrate away from the no-so-actively maintaned cibuilds/github.

What base image would you recommend? I've seen `ubuntu-1604` used here and there.

I like the approach of using sub-containers, by the way. It keeps things separated, and tidy. And, a volume can be used for intermediate storage into the same job.

Reply:
> What base image would you recommend? I've seen ubuntu-1604 used here and there.

Some reasonably modern Go or Rust image is usually good. We only use ubuntu-1604 for [machine executors](https://circleci.com/docs/2.0/executor-types/#using-machine) where [choices are limited and there was no ubuntu-2004 when we started](https://circleci.com/docs/2.0/configuration-reference/#available-machine-images).

However, the way you have it not is fine as well.

Reply:
Please take a look at the latest version. I added a helper shell script, instead of having it all packed into the docker entry point.